airflow/utils/log/secrets_masker.py

```diff
@@ -0,0 +1,221 @@
+        # all the default ones!
+
+        record = logging.getLogRecordFactory()(
+            # name, level, pathname, lineno, msg, args, exc_info, func=None, sinfo=None,
+            "x",
+            logging.INFO,
+            __file__,
+            1,
+            "",
+            tuple(),
+            exc_info=None,
+            func="funcname",
+        )
+        return frozenset(record.__dict__.keys()) - frozenset(('msg', 'args'))
+
+    def filter(self, record) -> bool:
+        if self.ALREADY_FILTERED_FLAG in record.__dict__:
+            # Filters are attached to multiple handlers and logs, keep a
+            # "private" flag that stops us needing to process it more than once
+            return True
+
+        if self.replacer:
+            for k, v in record.__dict__.items():
+                if k in self._record_attrs_to_ignore:
+                    continue
+                record.__dict__[k] = self.redact(v)
+            if record.exc_info:
+                exc = record.exc_info[1]
+                # I'm not sure if this is a good idea!
+                exc.args = (self.redact(v) for v in exc.args)
+        record.__dict__[self.ALREADY_FILTERED_FLAG] = True
+
+        return True
+
+    def _redact_all(self, item: "RedactableItem") -> "RedactableItem":
+        if isinstance(item, dict):
+            return {dict_key: self._redact_all(subval) for dict_key, subval in item.items()}
+        elif isinstance(item, str):
+            return '***'
+        elif isinstance(item, (tuple, set)):
+            # Turn set in to tuple!
+            return tuple(self._redact_all(subval) for subval in item)
+        elif isinstance(item, (Iterable, list)):
+            return list(self._redact_all(subval) for subval in item)
+        else:
+            return item
+
+    # pylint: disable=too-many-return-statements
+    def redact(self, item: "RedactableItem", name: str = None) -> "RedactableItem":
+        """
+        Redact an any secrets found in ``item``, if it is a string.
+
+        If ``name`` is given, and it's a "sensitve" name (see
+        :func:`should_hide_value_for_key`) then all string values in the item
+        is redacted.
+
+        """
+        if name and should_hide_value_for_key(name):
+            return self._redact_all(item)
+
+        if isinstance(item, dict):
+            return {dict_key: self.redact(subval, dict_key) for dict_key, subval in item.items()}
+        elif isinstance(item, str):
+            if self.replacer:
+                # We can't replace specific values, but the key-based redacting
+                # can still happen, so we can't short-circuit, we need to walk
+                # the strucutre.
+                return self.replacer.sub('***', item)
+            return item
+        elif isinstance(item, (tuple, set)):
+            # Turn set in to tuple!
+            return tuple(self.redact(subval) for subval in item)
+        elif isinstance(item, (Iterable, list)):
```

Comment:
`list` is iterable by default so `isinstance(item, Iterable)` is sufficient.

Reply:
Oh yes